examples/material-next-app-router-ts/src/components/Theme/ThemeRegistry/ThemeRegistry.tsx

```diff
@@ -0,0 +1,19 @@
+'use client';
+
+import * as React from 'react';
+import { ThemeProvider } from '@mui/material/styles';
+import CssBaseline from '@mui/material/CssBaseline';
+import { NextAppDirEmotionCacheProvider } from '@/components/Theme/ThemeRegistry/EmotionCache';
+
+import theme from './theme';
+
+export default function ThemeRegistry({ children }: { children: React.ReactNode }) {
+  return (
+    <React.Fragment>
+      <CssBaseline />
+      <NextAppDirEmotionCacheProvider options={{ key: 'mui' }}>
+        <ThemeProvider theme={theme}>{children}</ThemeProvider>
```

Comment:
I believe `<CssBaseline />` should be nested under `ThemeProvider`, so that dark mode works correctly.

Many thanks for this PR btw!

Reply:
I actually have encountered an issue:

If I change the theme's palette mode at runtime. The `Paper` and `Button` components do reflect the change, but others like `Typography` or `CssBaseline` do not. 

If I comment out `NextAppDirEmotionCacheProvider` changing the mode at runtime works.

Reply:
Could you please share the sandbox link?

Reply:
@kernelwhisperer I am facing the same problem .Is there a way to reflect baseline changes without CssThemeVars

Reply:
Agree, the `CssBaseline` should be under the `ThemeProvider`.